packages/synapse-react-client/src/components/EntityCitation/EntityCitation.test.tsx

```diff
@@ -0,0 +1,279 @@
+  })
+
+  it('renders "Cite project" when only project DOI exists', async () => {
+    mockUseGetDOIAssociation.mockImplementation((request, options) => {
+      if (request.id === 'syn61841662' && request.version === 1) {
+        return useGetEntityDOIMockImpl()
+      } else if (
+        request.id === 'syn60582629' &&
+        request.version === undefined
+      ) {
+        return useGetProjectDOIMockImpl()
+      }
+      return useGetEntityDOIMockImpl()
+    })
+
+    render(
+      <EntityCitation
+        projectId="syn60582629"
+        entityId="syn61841662"
+        versionNumber={1}
+      />,
+      { wrapper: createWrapper() },
+    )
+
+    // Set mock responses after hooks have initialized
+    act(() => {
+      setMockUseGetEntityBundleSuccess(fileWithDoiAssociation)
+      setEntityDOISuccess(null)
+      setProjectDOISuccess(doiProjectSuccess)
+    })
+
+    await openPopover('Cite project')
+
+    await screen.findByRole('dialog', {
+      name: /Citation options/i,
+    })
+  })
+
+  it('Both project and entity have DOIs', async () => {
+    render(
+      <EntityCitation
+        projectId="syn60582629"
+        entityId="syn61841662"
+        versionNumber={1}
+      />,
+      { wrapper: createWrapper() },
+    )
+
+    act(() => {
+      setMockUseGetEntityBundleSuccess(fileWithDoiAssociation)
+      setEntityDOISuccess(doiEntitySuccess)
+      setProjectDOISuccess(doiProjectSuccess)
+    })
+
+    await openPopover('Cite as...')
+
+    await waitFor(() => {
+      const citePageButton = screen.getByRole('menuitem', {
+        name: /cite only this page/i,
+      })
+      expect(
+        screen.getByRole('menuitem', { name: /cite this project/i }),
+      ).toBeInTheDocument()
+      expect(citePageButton).toBeInTheDocument()
+      userEvent.click(citePageButton)
+    })
+
+    await screen.findByRole('dialog', {
+      name: /Citation options/i,
+    })
+  })
+
+  it('Versioned Entity DOI', async () => {
+    mockUseGetDOIAssociation.mockImplementation((request, options) => {
+      if (request.id === 'syn66268092' && request.version === 1) {
+        return useGetEntityDOIMockImpl()
+      } else if (
+        request.id === 'syn60582629' &&
+        request.version === undefined
+      ) {
+        return useGetProjectDOIMockImpl()
+      }
+      return useGetEntityDOIMockImpl()
+    })
+
+    render(
+      <EntityCitation
+        projectId="syn60582629"
+        entityId="syn66268092"
+        versionNumber={1}
+      />,
+      { wrapper: createWrapper() },
+    )
+
+    act(() => {
+      setMockUseGetEntityBundleSuccess(fileWithDoiAssociation)
+      setEntityDOISuccess(versionedDoiEntitySuccess)
+      setProjectDOISuccess(null)
+    })
+
+    await openPopover('Cite page')
+
+    await waitFor(() => {
+      screen.findByRole('dialog', {
+        name: /Citation options/i,
+      })
+      expect(
+        screen.getByText(content => content.includes('version=1')),
+      ).toBeInTheDocument()
+    })
```

Comment:
Either use `await screen.findBy..` outside of a `waitFor`, or use `getBy` in the `waitFor` (`findBy` is just a convenience wrapper over `await waitFor(() => getBy...)`)

In this case, `findBy` returns a promise that might not reject until `waitFor` has finished.

```suggestion
    await waitFor(() => {
      screen.getByRole('dialog', {
        name: /Citation options/i,
      })
      screen.getByText(content => content.includes('version=1'))
    })
```